From: Samuel Holland <samuel@sholland.org>
Date: Wed, 23 Sep 2020 00:13:54 -0500
Subject: [PATCH 205/465] arm64: dts: allwinner: pinephone: Enable internal
 HMIC bias

Revisions 1.0 and 1.1 of the PinePhone mainboard do not have an external
resistor connecting HBIAS to MIC2. Enable the internal resistor to
provide the appropriate mic bias.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.0.dts | 4 ++++
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.2.dts | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.0.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.0.dts
index 219f720..537c3ff 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.0.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.0.dts
@@ -18,3 +18,7 @@ &sgm3140 {
 	enable-gpios = <&pio 2 3 GPIO_ACTIVE_HIGH>; /* PC3 */
 	flash-gpios = <&pio 3 24 GPIO_ACTIVE_HIGH>; /* PD24 */
 };
+
+&codec_analog {
+	allwinner,internal-bias-resistor;
+};
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.2.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.2.dts
index 4e7e237..f12ea22 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.2.dts
@@ -35,6 +35,10 @@ &backlight {
 	default-brightness-level = <500>;
 };
 
+&codec_analog {
+	allwinner,internal-bias-resistor;
+};
+
 &lis3mdl {
 	/*
 	 * Board revision 1.2 fixed routing of the interrupt to DRDY pin,
